File: ExcelRibbon/LinksAnalyzer/ParseError.cs
using System;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExcelRibbon.LinksAnalyzer {
    public class ParseError {
        public ParseError(InternalCellRef cellRef, string formula, long charPosition, string condition) {
            CellRef = cellRef;
            Formula = formula;
            Condition = condition;
            CharPosition = charPosition;
        }

        public InternalCellRef CellRef {  get; }
        public string Formula      { get; }
        public long   CharPosition { get; }
        public string Condition    {  get; }
    }
}

File: LinksAnalyzerVBA/ParseError.vb
Public Class ParseError

    Private mCellRef As InternalCellRef
    Private mFormula As String
    Private mCondition As String
    Private mCharPosition As Long

    Friend Function Initialize(ByVal CellRef As InternalCellRef, ByVal Formula As String,
    ByVal CharPosition As Long,
    ByVal Condition As String
) As ParseError
    Set mCellRef = CellRef
    mFormula = Formula
        mCondition = Condition
        mCharPosition = CharPosition
    Set Initialize = Me
End Function

    Friend Property Get This() As ParseError : Set This = Me:                  End Property

    Public Property Get CellRef() As InternalCellRef : Set CellRef = mCellRef:         End Property

    Public Property Get CharPosition() As Long :          CharPosition = mCharPosition:   End Property

    Public Property Get Condition() As String :           Condition = mCondition:         End Property

    Public Property Get Formula() As String :             Formula = mFormula:             End Property

End Class
